game/ato/flightplans/shiprecoverytanker.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Iterator, Type
+from game.ato.flightplans.standard import StandardFlightPlan, StandardLayout
+from game.ato.flightplans.ibuilder import IBuilder
+from game.ato.flightplans.standard import StandardLayout
+from game.ato.flightplans.waypointbuilder import WaypointBuilder
+from game.ato.flightwaypoint import FlightWaypoint
+from game.utils import feet
+
+
+@dataclass(frozen=True)
+class RecoveryTankerLayout(StandardLayout):
+    nav_to: list[FlightWaypoint]
+    recovery_ship: FlightWaypoint
+    nav_from: list[FlightWaypoint]
+
+    def iter_waypoints(self) -> Iterator[FlightWaypoint]:
+        yield self.departure
+        yield from self.nav_to
+        yield self.recovery_ship
+        yield from self.nav_from
+        yield self.arrival
+        if self.divert is not None:
+            yield self.divert
+        yield self.bullseye
+
+
+class RecoveryTankerFlightPlan(StandardFlightPlan[RecoveryTankerLayout]):
+    @staticmethod
+    def builder_type() -> Type[Builder]:
+        return Builder
+
+    @property
+    def tot_waypoint(self) -> FlightWaypoint:
+        return self.layout.recovery_ship
+
+    @property
+    def mission_departure_time(self) -> timedelta:
+        return timedelta(hours=2)
+
+    @property
+    def patrol_start_time(self) -> timedelta:
+        return self.package.time_over_target
+
+    @property
+    def patrol_end_time(self) -> timedelta:
+        return self.tot + self.mission_departure_time
+
+    def tot_for_waypoint(self, waypoint: FlightWaypoint) -> timedelta | None:
+        if waypoint == self.tot_waypoint:
+            return self.tot
+        return None
+
+    def depart_time_for_waypoint(self, waypoint: FlightWaypoint) -> timedelta | None:
+        if waypoint == self.tot_waypoint:
+            return self.tot + self.mission_departure_time
+        return None
+
+
+class Builder(IBuilder[RecoveryTankerFlightPlan, RecoveryTankerLayout]):
+    def layout(self) -> RecoveryTankerLayout:
+
+        # TODO: Propagate the ship position.
+        ship = self.package.target.position
+
+        builder = WaypointBuilder(self.flight, self.coalition)
+
+        recovery = builder.recovery_tanker(ship)
+
+        tanker_type = self.flight.unit_type
+        altitude = tanker_type.preferred_patrol_altitude
```

Comment:
Hmm. If the player tasks a KC-135 for this, the recovery tanker will be at 24k ft. Not useful. Not something they're likely to do either, but it doesn't seem like it's prevented. A couple options:

1. Make this a unique mission type so we can configure tasking allow/disallow separately, allow the S-3 (and whatever else is reasonable for a recovery tanker), but not the KC-135/KC-130, and make sure the recovery type tankers do it at a reasonable altitude (but that might make them worse for regular tanking missions).
2. Make this not aircraft specific, and low enough that you're not going to run out of fuel climbing back up to the gas station.

I'm not sure which behavior makes more sense, or if it should even do both (I have no idea if it's reasonable for a KC-130 to provide recovery tanking for a CVN). I do know that 24k ft is not a useful place to do it though :) WDYT?